cloud/google/cmd/gce-controller/cluster-controller-app/controller.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster_controller_app
+
+import (
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/apiserver-builder/pkg/controller"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/kubernetes"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/cluster-api/cloud/google"
+	"sigs.k8s.io/cluster-api/cloud/google/cmd/gce-controller/cluster-controller-app/options"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+	clusterapiclientsetscheme "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/scheme"
+	"sigs.k8s.io/cluster-api/pkg/controller/cluster"
+	"sigs.k8s.io/cluster-api/pkg/controller/config"
+	"sigs.k8s.io/cluster-api/pkg/controller/sharedinformers"
+)
+
+const (
+	gceClusterControllerName = "gce-cluster-controller"
+)
+
+func StartClusterController(server *options.ClusterControllerServer, shutdown <-chan struct{}) {
```

Comment:
this file is very similar to the machine controller version.
does it make sense to try to refactor some of this logic into a shared library.
i imagine if we change/fix something in one, we would do it for the other, especially for the leader election and recorder logic.

not saying you should do it in this PR, but a todo and/or issue would do.